### DISCUSSION TITLE
Utilising External vector db

### DISCUSSION THREAD

**Discussion by leaykar** [open-webui/open-webui#2600] [General]
Hey llama lovers - 

I'm curious if it is possible; or how to correctly utilise an external chromadb instance for the vector db.

I have defined this in my open webui via 'CHROMA_HTTP_HOST' ; and I can see that the vector db i have externally in a standalone container is being correctly populated with the collection data coming from open webui.

But my goal in the end is to build my vector db without using the functionality within open webui via my own external custom code or other tools.

The problem I'm facing however is actually then querying this correctly via open webui as there are no 'documents' available - I'm not able to use the '#docname' or #all references to actually then ensure my query is referencing the RAG context?

Is there some way to generically have every query that a particular custom model file configuration has to always just do some query against the vector db collections?

**Comment by justinh-rahb** [comment]
The WebUI's backend is intentionally designed to not use a ChromaDB if it's got existing collections, there's no way to manipulate the vector store externally if you want it to be used by WebUI. You'd need to at the very least use our RAG API endpoints with your own code.

**Reply by leaykar** [reply]
Oh I see, forgive my ignorance but is there some reason for that?

Is there any documentation on the api for RAG at all on that side? (/rag/api/v1 ?)

**Reply by justinh-rahb** [reply]
If I had to make an uneducated guess it's likely that we just can't count on other applications using the same data and cohabitating without interference and unexpected results. Safer to just keep our own exclusive vector-store.

**Reply by leaykar** [reply]
Yeah fair enough makes sense - I already have an external project that is building out a corpus of PDF's into more content defined chunks rather than just character based auto chunking - My goal would effectively be to write this into a chromadb as well that this would then be able to consume;

**Comment by jag89** [comment]
@leaykar  , can you share the docker container creation code for open web ui where you have defined CHROMA_HTTP_HOST . preferably both Chroma db docker code and open web ui docker code which has reference to chroma_http_host. thanks in advance !

**Reply by leaykar** [reply]
I use openshift for my orchestration, it's now effectively
Ollama deployment 
Milvus deployment (stand-alone)(milvus, minio, etcd)
Tika deployment 
Openwebui deployment

**Reply by jag89** [reply]
can you just let me know how you are using Chroma_htttp_host variable ?

**Reply by leaykar** [reply]
It should just be:
VECTOR_DB=chroma 
CHROMA_HTTP_HOST=http://(hostname)

Assuming local docker I'd gander this is whatever your docker network hostname is set to for that container. 

I only ever focused on openshift/kube so this was all defined via deployments and configmaps
